dll/win32/shell32/dialogs/filetypes.cpp

```diff
@@ -1134,34 +1301,43 @@ EditTypeDlg_WriteClass(HWND hwndDlg, PEDITTYPE_DIALOG pEditType,
             if (RegCreateKeyExW(hVerbKey, L"command", 0, NULL, 0, KEY_WRITE, NULL,
                                 &hCommandKey, NULL) == ERROR_SUCCESS)
             {
-                // write the default value
-                DWORD dwSize = (value.GetLength() + 1) * sizeof(WCHAR);
-                RegSetValueExW(hCommandKey, NULL, 0, REG_EXPAND_SZ, LPBYTE(LPCWSTR(value)), dwSize);
-
+                DWORD dwSize = (cmd.GetLength() + 1) * sizeof(WCHAR);
+                DWORD dwType = REG_SZ;
+                LPCWSTR exp;
+                if ((exp = StrChrW(cmd, '%')) && StrChrW(exp + 1, '%'))
+                    dwType = REG_EXPAND_SZ;
+                RegSetValueExW(hCommandKey, NULL, 0, dwType, LPBYTE(LPCWSTR(cmd)), dwSize);
                 RegCloseKey(hCommandKey);
             }
 
             RegCloseKey(hVerbKey);
         }
     }
 
-    // set class name to class key
-    RegSetValueExW(hClassKey, NULL, 0, REG_SZ, LPBYTE(ClassName), cchName);
-
     RegCloseKey(hShellKey);
     RegCloseKey(hClassKey);
-
     return TRUE;
 }
 
 static BOOL
-EditTypeDlg_ReadClass(HWND hwndDlg, PEDITTYPE_DIALOG pEditType, LPCWSTR ClassKey)
+EditTypeDlg_ReadClass(HWND hwndDlg, PEDITTYPE_DIALOG pEditType, EDITTYPEFLAGS &ETF)
 {
-    // open class key
+    PFILE_TYPE_ENTRY pEntry = pEditType->pEntry;
+    LPCWSTR ClassKey;
+    HRESULT hr = GetClassKey(*pEntry, ClassKey);
     HKEY hClassKey;
-    if (RegOpenKeyExW(HKEY_CLASSES_ROOT, ClassKey, 0, KEY_READ, &hClassKey) != ERROR_SUCCESS)
+    if (FAILED(hr) || RegOpenKeyExW(HKEY_CLASSES_ROOT, ClassKey, 0, KEY_READ, &hClassKey))
         return FALSE;
 
+    UINT etf = (RegValExists(hClassKey, L"AlwaysShowExt")) ? ETF_ALWAYSEXT : 0;
```

Comment:
Why not directly modifying what you've called the "ETF" (allcaps) variable?

Reply:
Because the compiler does not like it when you bitflip an enum.